### ISSUE TITLE
Issue with Dinner Tables

### ISSUE THREAD

**Issue by Nachichuri** [opensourcepos/opensourcepos#1933]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

- OSPOS version is: 3.1.1  
- OSPOS git commit hash is: 
- PHP version is: 7.0
- MySQL or MariaDB version is: MariaDB 15.1
- OS and version is: Debian 9 Stretch
- WebServer is: Apache/2.4.25
- Selected language is: American English
- (If applicable) Docker installation: No
- (If applicable) Installation package for the LAMP/LEMP stack is: 

### Issue / Bug  / Question / New Feature

I've enabled the Dinner Tables features in the Store Config, since I'm using OSPOS for a client that owns a bar.

**Expected behaviour:**
There's not a ton of documentation about it, yet I understand that if you start a sale in Sales, (sales register mode is by default) and you start adding items to Table X, as soon as you change to Table Y, you can add independent items to the list. 

**Issue:**
When I add an item to a certain table, it clones to every other table, making it impossible to have separate orders in each table.

[Here's a video where I show what happens.](https://youtu.be/LH1n4gggX0Y)

I really like OSPOS, but this is making it impossible to be viable for restaurants and such, since there have to be open tables with different items on it. I hope somebody can help.

**Comment by mckaygerhard** [comment]
errr, seems he have right!

**Comment by Nachichuri** [comment]
**UPDATE**

I kept trying it out and I saw how it actually works. You have to select the Table, then load the items and _suspend_ the sale to start with a new table.

You then have to go to the suspended button, and _unsuspend_ the table's sale to keep adding items and/or complete the sale.

It's a really awkward and counter-intuitive method, but I guess it works. If you could work a bit more on it, and make it less of a fuss, Open Socurce POS would definitely be a solid POS alternative for restaurants.

**Comment by nerv55** [comment]
The table then is just act like label for the transaction, I have just reproduce in my installation what i have see in the video you posted.
may be when you change the table while the transaction is not finish the document get auto suspended and the when you change table back the transaction resume until the user finished, will be more friendly for the user.

**Comment by Nachichuri** [comment]
> The table then is just act like label for the transaction, I have just reproduce in my installation what i have see in the video you posted.
may be when you change the table while the transaction is not finish the document get auto suspended and the when you change table back the transaction resume until the user finished, will be more friendly for the user. @nerv55 

That's indeed a good alternative. Just making it a bit more intuitive, regardless of the process behind.

**Comment by jlctmaster** [comment]
That would be an improvement for that process, because when I was working on this module I was just finishing what someone else started and could not finish.

**Comment by nerv55** [comment]
@Nachichuri when you suspend the transaction the table involved disappear from you select List?

**Comment by jlctmaster** [comment]
That's right

**Comment by nerv55** [comment]
I'm asking because when I unsuspend the transaction my table don't come back in the list they just disappear and only see two first options.

**Comment by Nachichuri** [comment]
@nerv55 yes it does. Yet to add more items or finish the sale, you have to un-suspend it, which is a bit awkward and could definitely be improved.

**Comment by nerv55** [comment]
by the way I have 3.2.0

**Comment by Nachichuri** [comment]
@nerv55 that's because there's a shady and very discrete "Suspended" button you have to click to unsuspend the order.

**Comment by jlctmaster** [comment]
It will be a matter of applying the actions of the suspend button as an event when changing the table in the list

**Comment by Nachichuri** [comment]
@jlctmaster I'd love to see that done! :)

**Comment by nerv55** [comment]
@Nachichuri , yes I have  use the button to suspend the transactions, and the table involved is not present any more in tables list, it's ok to this point, but when I un-suspend the transaction the table I have selected does not come back, neither in transaction and in select list. again i have 3.2.0 and yours is 3.1.1, I just wan't to know if you have the same behavior

**Comment by Nachichuri** [comment]
@nerv55 well that sounds more as a bug my friend, when I unsuspend it works like a charm...

**Comment by nerv55** [comment]
ok, then don't upgrate to 3.2.0

**Comment by jlctmaster** [comment]
let me prepare the development environment again with this system, since I bought a new laptop and this is preparing it. ;)

**Comment by Nachichuri** [comment]
@jlctmaster you're my hero.

**Comment by nerv55** [comment]
I don't have 3.1.1, @Nachichuri  can you post zip with your html folder and sql file with your database struncture only, I can look into too

**Comment by Nachichuri** [comment]
@nerv55 I don't really know how compromising that'd be my friend, yet if you're looking for the server files, [here they are](https://github.com/opensourcepos/opensourcepos/releases/download/3.1.1/opensourcepos.20170909085348.3.1.1.76a002.zip), and I followed [these instructions](http://www.opensourceposguide.com/guide/gettingstarted/installation).

**Comment by jlctmaster** [comment]
Please change the issue title to "improve the Dinner Table module"

**Comment by mckaygerhard** [comment]
@Nachichuri @jlctmaster @nerv55 this its a demostration of that a very VAGE README and a weak documentation are the main pilar on any project! i'll try to post all the information here in the wiki later, htanks to all, i dont see other working on this "problem"

**Comment by mckaygerhard** [comment]
Ready now: https://github.com/opensourcepos/opensourcepos/wiki/DOCS-USERS-Sales-Restaurant i revised the problem of the comment https://github.com/opensourcepos/opensourcepos/issues/1933#issuecomment-379501678 and seems it happened to me too i'll report as a bug right now

**Comment by WebShells** [comment]
Agree... It should be in separated view... Yet it does the job for time being... Hope in the future we can proceed in this module..

**Comment by daN4cat** [comment]
@mckaygerhard, hope @jlctmaster can review and fix the documentation.

**Comment by mckaygerhard** [comment]
umm hi @daN4cat seems you close https://github.com/opensourcepos/opensourcepos/issues/1933#issuecomment-379600903 and do not try as @nerv55 suggests

about documentation i refers to the wiki, now structured but the API docs that was linked in old wiki text today does not work, well  ihope @jekkos see too and make some opinion about.. it not finished due upload a picture it's very dificult due only permits url images and not uploads

**Comment by mckaygerhard** [comment]
umm hi @daN4cat seems you close https://github.com/opensourcepos/opensourcepos/issues/1934 and do not try as @nerv55 suggests

about documentation i refers to the wiki, now structured but the API docs that was linked in old wiki text today does not work, well  ihope @jekkos see too and make some opinion about.. it not finished due upload a picture it's very dificult due only permits url images and not uploads

**Comment by daN4cat** [comment]
There is already this ticket, no need to create more.
Somebody needs to take and improve this feature. That somebody is not me and by the way this will not block 3.2.0 release.

**Comment by mckaygerhard** [comment]
the fact its that you said "it's working", seems not but thanks so then this issue encolse that!
